File: GeneradorPAXVB/GeneradorPAXVB/AccesoDisco.vb
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.IO

Public Class AccesoDisco

#Region "Métodos Públicos"

    Public Shared Function RecuperaListaArchivos(ByVal directorioRaiz As String) As IList
        Dim listaArchivos = Directory.GetFiles(directorioRaiz).ToList()
        Return listaArchivos
    End Function

    Public Shared Function RecuperaArchivos(ByVal rutaAbsoluta As String) As Stream
        Return File.OpenRead(rutaAbsoluta)
    End Function

    Public Shared Sub MoverArchivo(ByVal rutaAbsoluta As String, ByVal rutaDestino As String)
        Dim nombreArchivo = Path.GetFileName(rutaAbsoluta)
        File.Move(rutaAbsoluta, String.Format("{0}\\{1}", rutaDestino, nombreArchivo))
    End Sub

    Public Shared Sub GuardarArchivoLog(ByVal rutaAbsoluta As String, ByVal contenidoArchivo As List(Of String))
        File.WriteAllLines(rutaAbsoluta, contenidoArchivo.ToArray())
    End Sub

    Public Shared Sub GuardarArchivoTexto(ByVal rutaAbsoluta As String, ByVal contenidoArchivo As String)
        File.WriteAllText(rutaAbsoluta, contenidoArchivo)
    End Sub

#End Region

End Class

File: SolucionPruebas/SolucionPruebas.Presentacion.WindowsForms/AccesoDisco.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections;
using System.IO;

namespace SolucionPruebas.Presentacion.WindowsForms
{
    class AccesoDisco
    {
        #region Métodos Públicos
        public static IList RecuperaListaArchivos(string directorioRaiz)
        {
            IList listaArchivos = Directory.GetFiles(directorioRaiz).ToList();
            return listaArchivos;
        }

        public static Stream RecuperaArchivo(string rutaAbsoluta)
        {
            return File.OpenRead(rutaAbsoluta);
        }

        public static void MoverArchivo(string rutaAbsoluta, string rutaDestino)
        {
            var nombreArchivo = Path.GetFileName(rutaAbsoluta);
            File.Move(rutaAbsoluta, string.Format("{0}\\{1}", rutaDestino, nombreArchivo));
        }

        public static void GuardarArchivoLog(string rutaAbsoluta, List<string> contenidoArchivo)
        {
            File.WriteAllLines(rutaAbsoluta, contenidoArchivo.ToArray());
        }

        public static void GuardarArchivoTexto(string rutaAbsoluta, string contenidoArchivo)
        {
            File.WriteAllText(rutaAbsoluta, contenidoArchivo);
        }
        #endregion
    }
}
